src/gui/editors/Editor.cpp

```diff
@@ -89,51 +89,51 @@ void Editor::toggleMaximize()
 	isMaximized() ? showNormal() : showMaximized();
 }
 
-Editor::Editor(bool record, bool stepRecord) :
-	m_toolBar(new DropToolBar(this)),
-	m_playAction(nullptr),
-	m_recordAction(nullptr),
-	m_recordAccompanyAction(nullptr),
-	m_toggleStepRecordingAction(nullptr),
-	m_stopAction(nullptr)
+Editor::Editor(bool record, bool recordAccompany, bool recordStep) :
+    m_toolBar(new DropToolBar(this)),
+    m_playAction(nullptr),
+    m_recordAction(nullptr),
+    m_recordAccompanyAction(nullptr),
+    m_toggleStepRecordingAction(nullptr),
+    m_stopAction(nullptr)
 {
-	m_toolBar = addDropToolBarToTop(tr("Transport controls"));
-
-	auto addButton = [this](QAction* action, QString objectName) {
-		m_toolBar->addAction(action);
-		m_toolBar->widgetForAction(action)->setObjectName(objectName);
-	};
-
-	// Set up play and record actions
-	m_playAction = new QAction(embed::getIconPixmap("play"), tr("Play (Space)"), this);
-	m_stopAction = new QAction(embed::getIconPixmap("stop"), tr("Stop (Space)"), this);
-
-	m_recordAction = new QAction(embed::getIconPixmap("record"), tr("Record"), this);
-	m_recordAccompanyAction = new QAction(embed::getIconPixmap("record_accompany"), tr("Record while playing"), this);
-	m_toggleStepRecordingAction = new QAction(embed::getIconPixmap("record_step_off"), tr("Toggle Step Recording"), this);
-
-	// Set up connections
-	connect(m_playAction, SIGNAL(triggered()), this, SLOT(play()));
-	connect(m_recordAction, SIGNAL(triggered()), this, SLOT(record()));
-	connect(m_recordAccompanyAction, SIGNAL(triggered()), this, SLOT(recordAccompany()));
-	connect(m_toggleStepRecordingAction, SIGNAL(triggered()), this, SLOT(toggleStepRecording()));
-	connect(m_stopAction, SIGNAL(triggered()), this, SLOT(stop()));
-	new QShortcut(Qt::Key_Space, this, SLOT(togglePlayStop()));
-	new QShortcut(QKeySequence(combine(Qt::SHIFT, Qt::Key_Space)), this, SLOT(togglePause()));
-	new QShortcut(QKeySequence(combine(Qt::SHIFT, Qt::Key_F11)), this, SLOT(toggleMaximize()));
-
-	// Add actions to toolbar
-	addButton(m_playAction, "playButton");
-	if (record)
-	{
-		addButton(m_recordAction, "recordButton");
-		addButton(m_recordAccompanyAction, "recordAccompanyButton");
-	}
-	if(stepRecord)
-	{
-		addButton(m_toggleStepRecordingAction, "stepRecordButton");
-	}
-	addButton(m_stopAction, "stopButton");
+    m_toolBar = addDropToolBarToTop(tr("Transport controls"));
+
+    auto addButton = [this](QAction* action, QString objectName) {
+        m_toolBar->addAction(action);
+        m_toolBar->widgetForAction(action)->setObjectName(objectName);
+    };
+
+    auto createButton = [this, &addButton](
```

Comment:
`addButton` is now only ever used inside `createButton`, so instead of declaring it and passing to `createButton`, I would just copy the 2 code lines into `createButton`?